src/compiler/path.ts

```diff
@@ -475,9 +478,9 @@ namespace ts {
                 }
                 else if (reduced[0]) continue;
             }
-            reduced.push(component);
+            reduced?.push(component);
         }
-        return reduced;
+        return reduced ?? components as string[];

```

Comment:
What if your return type was also a `readonly string[]`? Is anyone really modifying the output?

Reply:
Yes, it broke stuff.